### ISSUE TITLE
Tables Wrecked on Save When Escaped Backticks are Present

### ISSUE THREAD

**Issue by bphunter1972** [yzhang-gh/vscode-markdown#1108]
<!--
⚠ Before proceeding:


Check FAQ:
https://github.com/yzhang-gh/vscode-markdown#faq

Read the reporting guide, especially when you encountered lag, unresponsiveness, or other performance issues:
https://markdown-all-in-one.github.io/docs/contributing/issue.html

Test the latest dev build if needed:
https://github.com/yzhang-gh/vscode-markdown/actions/workflows/main.yml?query=event%3Apush+is%3Asuccess
-->

## Problem

When a markdown file has a table, and inside of that table are [escaped backticks](https://www.markdownguide.org/basic-syntax/#escaping-backticks), upon saving the file the extension re-formats the table, pretty much destroying what was there before.


<!--
A clear and concise description of what you expected and what actually happened.
-->

## How to reproduce

Here's an example snippet that shows the issue. Enter the following into a markdown file:

```md
| Macro Name                     | Input  | Purpose                                                                                                                                                                                          |
| ------------------------------ | ------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `` `exp_stuff_not_x ``         |        |                                                                                                                                                                                                  |
| `` `exp_stuff_x ``             |        |                                                                                                                                                                                                  |
|                                | `goop` |                                                                                                                                                                                                  |
|                                | `goop` |                                                                                                                                                                                                  |
|                                | `goop` |                                                                                                                                                                                                  |
|                                | `goop` |                                                                                                                                                                                                  |
| `` `exp_stuff_at_value_cond `` |        | Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor `incididunt` ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris |
| `` `exp_stuff_at_value_cond `` |        | nisi ut aliquip ex ea commodo consequat. Duis aute `irure` dolor in reprehenderit in `voluptate` velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat            |

```

Now, hit Save. The extension re-formats the text to the following:

```md
| Macro Name                                                                                                                                                                                                                                 | Input  | Purpose |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------ | ------- |
| `` `exp_stuff_not_x ``                                                                                                                                                                                                                     |        |         |
| `` `exp_stuff_x ``                                                                                                                                                                                                                         |        |         |
|                                                                                                                                                                                                                                            | `goop` |         |
|                                                                                                                                                                                                                                            | `goop` |         |
|                                                                                                                                                                                                                                            | `goop` |         |
|                                                                                                                                                                                                                                            | `goop` |         |
| `` `exp_stuff_at_value_cond `` |        | Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor `incididunt` ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris |
| `` `exp_stuff_at_value_cond `` |        | nisi ut aliquip ex ea commodo consequat. Duis aute `irure` dolor in reprehenderit in `voluptate` velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat            |

```


## Configuration

<!--
Copy modified settings as JSON, and paste them into the code block below.

If you didn't change anything, delete the code block and say:
Default.
-->

```jsonc
Default
```

## Environment

Markdown All in One: v3.4.0
VS Code Version: 1.66.1 (Universal)
Commit: 8dfae7a5cd50421d10cd99cb873990460525a898
Date: 2022-04-06T14:49:47.400Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Darwin x64 21.3.0



## Diagnostic data

<!--
Usually some suspicious error messages from the Console of the Developer Tools.
If you're reporting a performance issue, please check the guide carefully.
-->
None found

**Comment by bphunter1972** [comment]
Thank you. Any chance this could just be released as a patch version? My company is, understandably, rather paranoid about such things.

**Comment by yzhang-gh** [comment]
released in v3.4.2

**Comment by bphunter1972** [comment]
Awesome!!
